apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SpannerDate.cs

```diff
@@ -0,0 +1,408 @@
+        {
+            if (days < 0 || days > MaxNumberOfDays)
+            {
+                throw new ArgumentOutOfRangeException(nameof(days));
+            }
+
+            _days = days;
+        }
+
+        ///<inheritdoc/>
+        public int CompareTo(SpannerDate other) => _days.CompareTo(other._days);
+
+        /// <summary>
+        /// Implementation of <see cref="IComparable.CompareTo"/> to compare two <see cref="SpannerDate"/> values.
+        /// </summary>
+        /// <remarks>
+        /// This uses explicit interface implementation to avoid it being called accidentally. The generic implementation should usually be preferred.
+        /// </remarks>
+        /// <exception cref="ArgumentException"><paramref name="obj"/> is non-null but does not refer to an instance of <see cref="SpannerDate"/>.</exception>
+        /// <param name="obj">The object to compare this value with.</param>
+        /// <returns>The result of comparing this value with <paramref name="obj"/>. <paramref name="obj"/> is null, this method returns a value greater than 0.
+        /// </returns>
+        int IComparable.CompareTo(object obj)
+        {
+            if (obj == null)
+            {
+                return 1;
+            }
+            return obj is SpannerDate date
+                ? CompareTo(date)
+                : throw new ArgumentException($"Object must be of type {nameof(SpannerDate)}", nameof(obj));
+        }
+
+        /// <inheritdoc/>
+        public bool Equals(SpannerDate other) => CompareTo(other) == 0;
+
+        /// <inheritdoc/>
+        public override bool Equals(object obj) => obj is SpannerDate other && Equals(other);
+
+        /// <inheritdoc/>
+        public override int GetHashCode() => _days.GetHashCode();
+
+        /// <summary>
+        /// Parses a textual representation as a <see cref="SpannerDate"/>.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// <paramref name="text"/> must be a representation of a date value which can be represented by <see cref="SpannerDate"/>
+        /// </para>
+        /// </remarks>
+        /// <param name="text">The text to parse. Must not be null.</param>
+        /// <returns>The parsed value.</returns>
+        /// <exception cref="FormatException">The value could not be parsed as a <see cref="SpannerDate"/>.</exception>
+        public static SpannerDate Parse(string text)
+        {
+            return Parse(text, CultureInfo.CurrentCulture);
+        }
+
+        /// <summary>
+        /// Parses a textual representation as a <see cref="SpannerDate"/>.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// <paramref name="text"/> must be a representation of a date value which can be represented by <see cref="SpannerDate"/>
+        /// </para>
+        /// </remarks>
+        /// <param name="text">The text to parse. Must not be null.</param>
+        /// <param name="formatProvider">The format provider.</param>
```

Comment:
It's worth noting that this can be null, and that that means the current culture is used.